record/record_test.go

```diff
@@ -399,13 +399,16 @@ func TestBasicRecover(t *testing.T) {
 }
 
 func TestComplexRecover(t *testing.T) {
-	// The first record will be blockSize * 3 bytes long. Since each block has
-	// a 6 byte header, the first record will roll over into 4 blocks.
+	// The first record will be blockSize * 3 bytes long. Since each block has a
+	// 7 byte header, the first record will roll over into 4 blocks.
 	records := [][]byte{
-		[]byte(strings.Repeat("a", blockSize*3)),
-		[]byte(strings.Repeat("b", blockSize-headerSize)),
-		[]byte(strings.Repeat("c", blockSize-headerSize)),
+		[]byte(strings.Repeat("a", blockSize*3)),          // Consume 3 whole blocks + 1 fractional block.
+		[]byte(strings.Repeat("b", 32733)),                // Consume the remainder of the 4th block.
```

Comment:
This magic 32733 number needs explaining. Can you make it equal to some formula?